src/Terrain/terrain.cpp

```diff
@@ -448,8 +448,8 @@ std::vector<int> Terrain::generate_macro_map(unsigned int size_x, unsigned int s
     NoiseGenerator ng = NoiseGenerator(numOctaves, persistance, 3);
 
     for (unsigned int i = 0; i < out.size(); i++){
-        auto [x,y,z] = sop(i,size_x,size_y,1);
-        auto p = ng.getValueNoise((double)x*spacing,(double)y*spacing);
+        auto xyz = sop(i,size_x,size_y,1);
```

Comment:
What's wrong with destructuring? Did you set your C++ version?

https://en.cppreference.com/w/cpp/language/structured_binding